### PR TITLE
Use log library  from go-kit

### PR DESCRIPTION
**Pull request by bharaththiruveedula**

### CODE REVIEW

**Comment by bharaththiruveedula** [comment]
@brian-brazil

**Comment by bharaththiruveedula** [comment]
seems like the tests are breaking because of https://github.com/prometheus/memcached_exporter/issues/67

File: main.go [hunk 1]
```diff
@@ -701,16 +707,16 @@ func (e *Exporter) parseStatsSettings(ch chan<- prometheus.Metric, statsSettings
...
-	return e.extractValueAndNewMetric(ch, desc, valueType, parseBool, stats, key, labelValues...)
+func (e *Exporter) parseBoolAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, stats map[string]string, key string, logger log.Logger, labelValues ...string) error {
+	return e.extractValueAndNewMetric(ch, desc, valueType, parseBool, stats, key, logger, labelValues...)
 }
 
-func (e *Exporter) extractValueAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, f func(map[string]string, string) (float64, error), stats map[string]string, key string, labelValues ...string) error {
-	v, err := f(stats, key)
+func (e *Exporter) extractValueAndNewMetric(ch chan<- prometheus.Metric, desc *prometheus.Desc, valueType prometheus.ValueType, f func(map[string]string, string, log.Logger) (float64, error), stats map[string]string, key string, logger log.Logger, labelValues ...string) error {
+	v, err := f(stats, key, logger)
```
Comment (brian-brazil) [inline main.go]: logger is available as e.logger, no need to pass it down.

File: main.go [hunk 2]
```diff
@@ -506,34 +512,34 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	c, err := memcache.New(e.address)
 	if err != nil {
 		ch <- prometheus.MustNewConstMetric(e.up, prometheus.GaugeValue, 0)
-		log.Errorf("Failed to connect to memcached: %s", err)
+		level.Error(e.logger).Log("Failed to connect to memcached: %s", err)
```
Comment (brian-brazil) [inline main.go]: It's a key-value system so this should be `"msg", "Failed to connect to memcached", "err", err`

Comment (bharaththiruveedula) [reply main.go]: done

File: main.go (hunk 1 above)
Comment (bharaththiruveedula) [reply main.go]: Done

Comment (brian-brazil) [reply main.go]: This is still around in several places.

**Comment by brian-brazil** [comment]
:+1: 

@grobie I'll live the final review up to you.

**Comment by bharaththiruveedula** [comment]
@grobie can you please review?

File: main.go [hunk 3]
```diff
@@ -722,25 +728,25 @@ func (e *Exporter) extractValueAndNewMetric(ch chan<- prometheus.Metric, desc *p
...
 	if !ok {
-		log.Debugf("Key not found: %s", key)
+		level.Debug(logger).Log("msg", "Key not found", "key", key)
 		return 0, errKeyNotFound
 	}
 
 	v, err := strconv.ParseFloat(value, 64)
 	if err != nil {
-		log.Errorf("Failed to parse %s %q: %s", key, value, err)
+		level.Error(logger).Log("msg", "Failed to parse ", "key", key, "value", value, "err", err)
```
Comment (grobie) [inline main.go]:
```suggestion
		level.Error(logger).Log("msg", "Failed to parse", "key", key, "value", value, "err", err)
```

File: main.go [hunk 4]
```diff
@@ -750,7 +756,7 @@ func parseBool(stats map[string]string, key string) (float64, error) {
 	case "no":
 		return 0, nil
 	default:
-		log.Errorf("Failed parse %s %q", key, value)
+		level.Error(logger).Log("msg", "Failed parse ", "key", key, "value", value)
```
Comment (grobie) [inline main.go]:
```suggestion
		level.Error(logger).Log("msg", "Failed to parse", "key", key, "value", value)
```

File: main.go [hunk 5]
```diff
@@ -824,6 +831,10 @@ func main() {
              </body>
              </html>`))
 	})
-	log.Infoln("Starting HTTP server on", *listenAddress)
-	log.Fatal(http.ListenAndServe(*listenAddress, nil))
+
+	level.Info(logger).Log("msg", "Listening on address", "address", *listenAddress)
```
Comment (grobie) [inline main.go]: What's the motivation to remove the information that the HTTP server is being started? Minor thing, but `msg="Listening on address", address="localhost:80"` stutters.

File: main.go [hunk 6]
```diff
@@ -824,6 +831,10 @@ func main() {
...
+	if err := http.ListenAndServe(*listenAddress, nil); err != nil {
+		level.Error(logger).Log("msg", "Error starting HTTP server", "err", err)
```
Comment (grobie) [inline main.go]:
```suggestion
		level.Error(logger).Log("msg", "Error running HTTP server", "err", err)
```

The error can happen a while after the startup as well.

File: main.go (hunk 5 above)
Comment (bharaththiruveedula) [reply main.go]: To just make it uniform across all the projects

File: main.go (hunk 4 above)
Comment (bharaththiruveedula) [reply main.go]: Done

File: main.go (hunk 3 above)
Comment (bharaththiruveedula) [reply main.go]: Done

**Comment by bharaththiruveedula** [comment]
@grobie can you please review

**Pull request merged**